drivers/counter/counter_mchp_xec.c

```diff
@@ -0,0 +1,420 @@
+
+	LOG_DBG("%p Counter alarm set to %u ticks", dev, ticks);
+
+	counter->control_reg.start = 1;
+
+	return 0;
+}
+
+
+static int counter_xec_cancel_alarm(struct device *dev, u8_t chan_id)
+{
+	volatile struct counter_xec_registers *counter =
+						COUNTER_XEC_REG_BASE(dev);
+	struct counter_xec_data *data = COUNTER_XEC_DATA(dev);
+
+	counter->control_reg.start = 0;
+	counter->int_enable_reg.int_enable = 0;
+
+	data->alarm_cb = NULL;
+	data->user_data = NULL;
+
+	LOG_DBG("%p Counter alarm canceled", dev);
+
+	return 0;
+}
+
+static u32_t counter_xec_get_pending_int(struct device *dev)
+{
+	volatile struct counter_xec_registers *counter =
+						COUNTER_XEC_REG_BASE(dev);
+
+	return counter->status_reg.event_int;
+}
+
+static u32_t counter_xec_get_top_value(struct device *dev)
+{
+	volatile struct counter_xec_registers *counter =
+						COUNTER_XEC_REG_BASE(dev);
+	const struct counter_xec_config *counter_cfg = COUNTER_XEC_CONFIG(dev);
+
+	if (counter_cfg->info.max_top_value == UINT16_MAX) {
+		return (u32_t)counter->preload_reg.preload16;
+	}
+
+	return counter->preload_reg.preload32;
+}
+
+static int counter_xec_set_top_value(struct device *dev,
+				     const struct counter_top_cfg *cfg)
+{
+	volatile struct counter_xec_registers *counter =
+						COUNTER_XEC_REG_BASE(dev);
+	const struct counter_xec_config *counter_cfg = COUNTER_XEC_CONFIG(dev);
+	struct counter_xec_data *data = COUNTER_XEC_DATA(dev);
+	int ret = 0;
+
+	if (data->alarm_cb) {
+		return -EBUSY;
+	}
+
+	if (cfg->ticks > counter_cfg->info.max_top_value) {
+		return -EINVAL;
+	}
+
+	counter->control_reg.start = 0;
+
+	if (cfg->flags & COUNTER_TOP_CFG_DONT_RESET) {
+		if (counter->count_reg > cfg->ticks) {
+			ret = -ETIME;
+
+			if (cfg->flags & COUNTER_TOP_CFG_RESET_WHEN_LATE) {
+				counter->count_reg = cfg->ticks;
+			}
+		}
+	} else {
+		counter->count_reg = cfg->ticks;
+	}
+
+	if (counter_cfg->info.max_top_value == UINT16_MAX) {
+		counter->preload_reg.preload16 = cfg->ticks;
+	} else {
+		counter->preload_reg.preload32 = cfg->ticks;
+	}
+
+	data->top_cb = cfg->callback;
+	data->user_data = cfg->user_data;
+
+	if (data->top_cb) {
+		counter->int_enable_reg.int_enable = 1;
+		counter->control_reg.auto_restart = 1;
+	} else {
+		counter->int_enable_reg.int_enable = 0;
+		counter->control_reg.auto_restart = 0;
+	}
+
+	LOG_DBG("%p Counter top value was set to %u", dev, cfg->ticks);
+
+	counter->control_reg.start    = 1;
+
+	return ret;
+}
+
+static u32_t counter_xec_get_max_relative_alarm(struct device *dev)
+{
+	const struct counter_xec_config *counter_cfg = COUNTER_XEC_CONFIG(dev);
+
+	return counter_cfg->info.max_top_value;
+}
+
+static void counter_xec_isr(struct device *dev)
+{
+	volatile struct counter_xec_registers *counter =
+						COUNTER_XEC_REG_BASE(dev);
+	const struct counter_xec_config *counter_cfg = COUNTER_XEC_CONFIG(dev);
+	struct counter_xec_data *data = COUNTER_XEC_DATA(dev);
+	u32_t girq_result;
+
+	girq_result = MCHP_GIRQ_RESULT(counter_cfg->girq_id);
```

Comment:
Same probelm as in the WDT. You are using bit 0,1,4,5 from GIRQ23 and you are clearing other bits. If you leave the code as it is, this isr will clear bits from the RTOS timer too.  Change it to clear only the appropiate instances

Reply:
@scottwcpg Can you correct me about this just to be sure

Reply:
Francisco you are correct.  The RESULT register is the bit-wise OR of ENABLE and SOURCE.  Writing the value from RESULT to SOURCE will clear all bits that are active when RESULT is read.  This driver will clear interrupt source status that don't belong to it.  The RESULT value should be masked with a bit-mask for the hardware the driver is handling.